dll/win32/shell32/CDefView.cpp

```diff
@@ -1947,15 +1958,18 @@ LRESULT CDefView::DoColumnContextMenu(LPARAM lParam)
 */
 UINT CDefView::GetSelections()
 {
-    SHFree(m_apidl);
-
-    m_cidl = m_ListView.GetSelectedCount();
-    m_apidl = static_cast<PCUITEMID_CHILD*>(SHAlloc(m_cidl * sizeof(PCUITEMID_CHILD)));
-    if (!m_apidl)
+    UINT count = m_ListView.GetSelectedCount();
+    if (count > m_cidl || !count || !m_apidl) // !count to free possibly large cache, !m_apidl to make sure m_apidl is a valid pointer
```

Comment:
If the count is zero, then you would try to do an SHAlloc(0) below? Does this fail gracefully without asserts?

Reply:
So the SHAlloc/CoTaskMemAlloc/IMalloc_Alloc/[IMalloc_fnAlloc](https://git.reactos.org/?p=reactos.git;a=blob;f=dll/win32/ole32/ifs.c;hb=2cdd5eca7b6459df6f8139e1d85879fb4f91e6d2#l174)/HeapAlloc (which is its actual implementation in ROS) with size zero bytes does an actual allocation? That's interesting (and quite unexpected).
Side-note: the behaviour of `malloc(0)` is implementation-defined, see https://stackoverflow.com/a/2022402/21852502

Reply:
OK let's go with the possibility of zero-size allocation (indeed old code did this, just fetching the GetSelectedCount() value into SHAlloc), and hope for the best, since there's not going to be any "regression" compared to old code :joy: